vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -922,26 +905,25 @@ def check(self, instance):
 
             # Second part: do the job
             self.collect_metrics(instance)
-        else:
-            self.log.debug("Thread pool is still processing jobs from previous run. Not scheduling anything.")
 
-        self._query_event(instance)
+            self._query_event(instance)
 
-        thread_crashed = False
-        try:
-            while True:
-                self.log.error(self.exceptionq.get_nowait())
-                thread_crashed = True
-        except Empty:
-            pass
+            try:
+                thread_crashed = False
+                while True:
+                    # Pop every exceptions from the queue and log them
+                    self.log.error(self.exceptionq.get_nowait())
+                    thread_crashed = True
+            except Empty:
+                pass
 
-        if thread_crashed:
-            self.stop_pool()
-            raise Exception("One thread in the pool crashed, check the logs")
+            if thread_crashed:
+                raise Exception("One thread in the pool crashed, check the logs")
 
-        if set_external_tags is not None:
-            set_external_tags(self.get_external_host_tags())
+            if set_external_tags is not None:
+                set_external_tags(self.get_external_host_tags())
```

Comment:
note: since this runs `set_external_tags` after the collection is complete (instead of during the collection), these host tags may get sent later.

Reply:
actually all the host tags are populated by the call to `self._cache_morlist_raw`, which is now blocking, so my comment isn't relevant